.github/workflows/main.yaml

```diff
@@ -216,7 +216,7 @@ jobs:
           echo -e "Exported values:\nQASE_RUN_ID=${ID}\nQASE_RUN_DESCRIPTION=${QASE_RUN_DESCRIPTION}\nQASE_RUN_NAME=${QASE_RUN_NAME}"
 
       - name: Provisioning cluster tests
-        if:  ${{ steps.prepare-rancher.outcome == 'success' && contains(inputs.tests_to_run, 'p0_provisioning') }}
+        if: ${{ !cancelled() && steps.prepare-rancher.outcome != 'failure' && contains(inputs.tests_to_run, 'p0_provisioning') }}
```

Comment:
nit:
>possible outcome values are: success, failure, cancelled, or skipped, I guess != 'failure' should work here. (ref: https://github.com/rancher/hosted-providers-e2e/pull/219)

Shouldn't `prepare-rancher` always succeed for tests to run?
Thinking out loud, we are using `!cancelled() && steps.prepare-rancher.outcome != 'failure'`, we know that the step won't be skipped, so it would only run the tests if the step were a success, but I think it would be better to use `==success`, no?

I guess we can wait and see if this misbehaves, and then make changes.

Reply:
With `success` we won't be able to test on existing self-hosted rancher. I was thinking to use `success OR skipped` but in the end `not failure` covers both and `cancelled` is covered by extra `cancelled()` so this is like optimized approach.